server/src/main/java/org/jboss/as/server/deployment/module/ModuleDependency.java

```diff
@@ -101,6 +137,10 @@ public ModuleIdentifier getIdentifier() {
         return identifier;
     }
 
+    public String getReason() {
+        return reason;
+    }
```

Comment:
According to the discussion taken in https://github.com/wildfly/wildfly-core/pull/5353#discussion_r1085329304 this could return an Optional<String>. The difference is minimal, but since this can be used by any third party, getting an optional one is more friendly since advertises the optional nature of this field, especially if you are iterating over a stream of an object like this.  It is a nice to have.